pkgs/tools/text/mdbook-footnote/default.nix

```diff
@@ -0,0 +1,24 @@
+{ lib, stdenv, fetchFromGitHub, rustPlatform, CoreServices }:
+rustPlatform.buildRustPackage rec {
+  pname = "mdbook-footnote";
+  version = "0.1.1";
+
+  src = fetchFromGitHub {
+    owner = "daviddrysdale";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-WUMgm1hwsU9BeheLfb8Di0AfvVQ6j92kXxH2SyG3ses=";
+  };
+
+  cargoSha256 = "sha256-Ig+uVCO5oHIkkvFsKiBiUFzjUgH/Pydn4MVJHb2wKGc=";
+
+  buildInputs = lib.optionals stdenv.isDarwin [ CoreServices ];
+
+  meta = with lib; {
+    description =
+      "A preprocessor for mdbook to support the inclusion of automatically numbered footnotes.";
+    homepage = "https://github.com/daviddrysdale/mdbook-footnote";
+    license = [ licenses.asl20 ];
+    maintainers = with maintainers; [ brianmcgillion ];
```

Comment:
If the program results in a binary that is expected to be run you should include the meta tag `mainProgram`.

Reply:
It is a preprocessor plugin, that is registered in the `book.toml` and called during a run of `mdbook`, for your documentation project